contracts/JoepegExchange.sol

```diff
@@ -560,6 +594,53 @@ contract JoepegExchange is
         }
     }
 
+    /**
+     * @notice Match multiple asks with their respective taker bid order using AVAX and WAVAX, ignoring expired maker asks
+     * @dev Used when the caller doesn't want the transaction to revert if a maker ask already expired
+     * @param trades an array of trades
+     */
+    function batchBuyWithAVAXAndWAVAXIgnoringExpiredAsks(
+        Trade[] calldata trades
+    ) external payable nonReentrant {
+        // Calculate the total cost of all valid orders
+        uint256 totalCost;
+        for (uint256 i; i < trades.length; ++i) {
+            if (_validateMakerOrder(trades[i].makerAsk)) {
+                totalCost += trades[i].takerBid.price;
+            }
+        }
+
+        // Transfer WAVAX if needed
+        if (totalCost > msg.value) {
+            IERC20(WAVAX).safeTransferFrom(
+                msg.sender,
+                address(this),
+                (totalCost - msg.value)
+            );
+
+            // Wrap AVAX sent to this contract
+            IWAVAX(WAVAX).deposit{value: msg.value}();
+        } else {
+            // Wrap AVAX needed to pay for valid orders
+            IWAVAX(WAVAX).deposit{value: totalCost}();
+        }
+
+        // Match orders
+        for (uint256 i; i < trades.length; ++i) {
+            _matchAskWithTakerBidUsingAVAXAndWAVAXIgnoringExpiredAsks(
+                trades[i].takerBid,
+                trades[i].makerAsk
+            );
+        }
+
+        // Return remaining AVAX (if any)
+        if (msg.value > totalCost) {
+            uint256 remainingAVAX = msg.value - totalCost;
+            (bool sent, ) = msg.sender.call{value: remainingAVAX}("");
+            require(sent, "Batch Buy: Failed to return remaining AVAX");
+        }
```

Comment:
one thing to note here is that if the caller is a contract rather than an EOA, they may not support receiving AVAX and this could error. one way to work around it would be to send the refund back in WAVAX. curious what @LouisMeMyself thinks though

Reply:
interesting; for what it's worth, that's how gem does it:

```solidity
function batchBuyWithETH(
        MarketRegistry.TradeDetails[] memory tradeDetails
    ) payable external nonReentrant {
        // execute trades
        _trade(tradeDetails);

        // return remaining ETH (if any)
        assembly {
            if gt(selfbalance(), 0) {
                let callStatus := call(
                    gas(),
                    caller(),
                    selfbalance(),
                    0,
                    0,
                    0,
                    0
                )
            }
        }
    }
```

Reply:
ok im down to keep refund as AVAX then. we can change in the future if needed

Reply:
Yeah, it's okay tbh.
I raised the issue with `JoepegAuctionHouse` because it could lock an NFT, while here the worse case is that a contract may not be able to use that function, which is okay